pkg/smartcontract/binding/generate.go

```diff
@@ -19,34 +19,48 @@ import (
 )
 
 const srcTmpl = `
-{{- define "METHOD" -}}
-// {{.Name}} {{.Comment}}
-func {{.Name}}({{range $index, $arg := .Arguments -}}
-	{{- if ne $index 0}}, {{end}}
-		{{- .Name}} {{.Type}}
-	{{- end}}) {{if .ReturnType }}{{ .ReturnType }} {
-	return neogointernal.CallWithToken(Hash, "{{ .NameABI }}", int(contract.{{ .CallFlag }})
-		{{- range $arg := .Arguments -}}, {{.Name}}{{end}}).({{ .ReturnType }})
-	{{- else -}} {
-	neogointernal.CallWithTokenNoRet(Hash, "{{ .NameABI }}", int(contract.{{ .CallFlag }})
-		{{- range $arg := .Arguments -}}, {{.Name}}{{end}})
-	{{- end}}
-}
-{{- end -}}
-// Code generated by neo-go contract generate-wrapper --manifest <file.json> --out <file.go> --hash <hash> [--config <config>]; DO NOT EDIT.
+// Code generated by neo-go contract generate-wrapper --manifest <file.json> --out <file.go> [--hash <hash>] [--config <config>]; DO NOT EDIT.
 
 // Package {{.PackageName}} contains wrappers for {{.ContractName}} contract.
 package {{.PackageName}}
 
 import (
 {{range $m := .Imports}}	"{{ $m }}"
 {{end}})
-
+{{if .Hash}}
 // Hash contains contract hash in big-endian form.
-const Hash = "{{ .Hash }}"
+var Hash = "{{ .Hash }}"
+{{end}}
+
+// Contract represents a smart contract with a dynamic hash.
+type Contract struct {
+    hash interop.Hash160
+}
+
+// NewContract creates a new instance of Contract with the given hash.
+func NewContract(hash interop.Hash160) *Contract {
+    return &Contract{hash: hash}
+}
```

Comment:
You need to emit this code only in case if hash is not provided. Put it inside `{{if .Hash}}`. If hash is provided, then the old behaviour should be supported.